### PR TITLE
Next to develop pre 2.2.0

### PR DESCRIPTION
**Pull request by glennmatthews**
Please review. :-)

### CODE REVIEW

File: nautobot/core/templatetags/helpers.py [hunk 1]
```diff
@@ -81,6 +81,24 @@ def hyperlinked_object(value, field="display"):
     return format_html("{}", display)
 
 
+@library.filter()
+@register.filter()
+def hyperlinked_email(value):
+    """Render an email address as a `mailto:` hyperlink."""
+    if value is None:
```
Comment (glennmatthews) [inline nautobot/core/templatetags/helpers.py:88]:
```suggestion
    if not value:
```

File: nautobot/core/templatetags/helpers.py [hunk 2]
```diff
@@ -81,6 +81,24 @@ def hyperlinked_object(value, field="display"):
...
+        return placeholder(value)
+    return format_html('<a href="mailto:{}">{}</a>', value, value)
+
+
+@library.filter()
+@register.filter()
+def hyperlinked_phone_number(value):
+    """Render a phone number as a `tel:` hyperlink."""
+    if value is None:
```
Comment (glennmatthews) [inline nautobot/core/templatetags/helpers.py:97]:
```suggestion
    if not value:
```

File: nautobot/core/tests/test_navigations.py [hunk 3]
```diff
@@ -63,8 +65,30 @@ def test_menu_item_attributes(self):
...
+                self.assertEqual(expected_perms[f"{tab_name}:{group_name}"], group_perms)
+                # if any group has no permissions restriction, then the tab has no permissions restriction
+                if expected_perms[tab_name] is None or not group_perms:
+                    expected_perms[tab_name] = None
+                else:
+                    expected_perms[tab_name] |= group_perms
+            self.assertEqual(expected_perms[tab_name], tab_details["permissions"])
+
 
-@tag("unit")
```
Comment (glennmatthews) [inline nautobot/core/tests/test_navigations.py:67]: Why? This removal looks wrong at a glance.

File: nautobot/extras/tests/integration/test_plugins.py
Comment (glennmatthews) [inline nautobot/extras/tests/integration/test_plugins.py:1]: Should rename this file to `test_apps.py`.

File: nautobot/extras/tests/test_plugins.py
Comment (glennmatthews) [inline nautobot/extras/tests/test_plugins.py:1]: Should rename to `test_apps.py`

File: nautobot/core/tests/test_navigations.py (hunk 3 above)
Comment (glennmatthews) [reply nautobot/core/tests/test_navigations.py:67]: Correct, AFAIK.

Comment (glennmatthews) [reply nautobot/core/tests/test_navigations.py:67]: And to be clear here we're using django's base `TestCase` class, not the Nautobot subclass thereof (which may itself be wrong?)

File: nautobot/extras/tests/integration/test_plugins.py
Comment (glennmatthews) [reply nautobot/extras/tests/integration/test_plugins.py:1]: Maybe that's deferred to the "developer-facing" plugin-to-apps renames coming in 2.3?

File: nautobot/docs/release-notes/version-2.2.md [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+
+This document describes all new features and changes in Nautobot 2.2.
+
+## Release Overview
+
+### Added
+
+#### Contact and Team Models ([#230](https://github.com/nautobot/nautobot/issues/230))
+
+Contact and Team are models that represent an individual and a group of individuals who can be linked to an object. Contacts and teams store the necessary information (name, phone number, email, and address) to uniquely identify and contact them. They are added to track ownerships of organizational entities and to manage resources more efficiently in Nautobot. Check out the documentation for [contact](../user-guide/core-data-model/extras/contact.md) and [team](../user-guide/core-data-model/extras/team.md). There is also a [user guide](../user-guide/feature-guides/contacts-and-teams.md) available on how to utilize these models.
```
Comment (gsnider2195) [inline nautobot/docs/release-notes/version-2.2.md:13]:
```suggestion
Contact and Team are models that represent an individual and a group of individuals who can be linked to an object. Contacts and teams store the necessary information (name, phone number, email, and address) to uniquely identify and contact them. They are added to track ownerships of organizational entities and to manage resources more efficiently in Nautobot. Check out the documentation for [Contact](../user-guide/core-data-model/extras/contact.md) and [Team](../user-guide/core-data-model/extras/team.md). There is also a [user guide](../user-guide/feature-guides/contacts-and-teams.md) available on how to utilize these models.
```

File: nautobot/docs/release-notes/version-2.2.md [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+
+#### Controller Model ([#3111](https://github.com/nautobot/nautobot/issues/3111))
+
+Controller models have been added to the `dcim` app. A Controller in Nautobot is an abstraction meant to represent network or SDN (Software-Defined Networking) controllers. These may include, but are not limited to, wireless controllers, cloud-based network management systems, and other forms of central network control mechanisms.
+
+For more details, refer to the user guide for a [`Controller` model](../user-guide/core-data-model/dcim/controller.md), a [`ControllerManagedDeviceGroup` model](../user-guide/core-data-model/dcim/controllermanageddevicegroup.md), or developer documentation for [controllers](../development/core/controllers.md).
```
Comment (gsnider2195) [inline nautobot/docs/release-notes/version-2.2.md:21]:
```suggestion
For more details, refer to the user guide for a [`Controller` model](../user-guide/core-data-model/dcim/controller.md), a [`ControllerManagedDeviceGroup` model](../user-guide/core-data-model/dcim/controllermanageddevicegroup.md), or developer documentation for [Controllers](../development/core/controllers.md).
```

File: nautobot/docs/release-notes/version-2.2.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
...
+
+#### DeviceFamily Model ([#3559](https://github.com/nautobot/nautobot/issues/3559))
+
+A [device family](../user-guide/core-data-model/dcim/devicefamily.md) represents a group of related [device types](../user-guide/core-data-model/dcim/devicetype.md). A device type can be optionally assigned to a device family. Each device family must have a unique name and may have a description assigned to it.
```
Comment (gsnider2195) [inline nautobot/docs/release-notes/version-2.2.md:25]:
```suggestion
A [Device Family](../user-guide/core-data-model/dcim/devicefamily.md) represents a group of related [Device Types](../user-guide/core-data-model/dcim/devicetype.md). A Device Type can be optionally assigned to a Device Family. Each Device Family must have a unique name and may have a description assigned to it.
```

**Review by gsnider2195 — APPROVED** [review]

**Pull request merged**